src/NHibernate.Test/NHSpecificTest/NH3932/Model.cs

```diff
@@ -252,7 +275,7 @@ public override bool Equals(object obj)
 
 		public override int GetHashCode()
 		{
-			return 1; //for simlicity - care about equals impl
+			return Id;
```

Comment:
First, thanks for your careful and quick reviews!

About the gethashcode impl in test model... IMHO, I think the test model should meet the requirements of the tests. They don't have to work for scenarios not (yet) present in the test.
(On the other hand they should not make tests false positive and they should also, to some extend, show "good" use case examples - which is more what you, correctly, pointing out).

I change it to simplest impl to always give a correct result (=hard coded value). Good enough?